exporter/otlp-common/lib/opentelemetry/exporter/otlp/common.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'opentelemetry/exporter/otlp/common/version'
+
+require 'google/rpc/status_pb'
+
+require 'opentelemetry/proto/common/v1/common_pb'
+require 'opentelemetry/proto/resource/v1/resource_pb'
+require 'opentelemetry/proto/trace/v1/trace_pb'
+require 'opentelemetry/proto/collector/trace/v1/trace_service_pb'
+
+module OpenTelemetry
+  module Exporter
+    module OTLP
+      # Contains common functionality between the different OTLP export protocols
+      module Common # rubocop:disable Metrics/ModuleLength
+        extend self
+
+        def encode(span_data) # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+          Opentelemetry::Proto::Collector::Trace::V1::ExportTraceServiceRequest.encode(as_etsr(span_data))
+        rescue StandardError => e
+          OpenTelemetry.handle_error(exception: e, message: 'unexpected error in OTLP::Exporter#encode')
+          nil
+        end
+
+        def as_etsr(span_data)
```

Comment:
Does this need to be public? For tests? Could we mark this API private with a comment?

Reply:
Maybe I'll merge the GRPC PR back into this one so it's easier to see. But GRPC handles the encoding/decoding itself, so we need to expose the proto format for the GRPC exporter.

https://github.com/open-telemetry/opentelemetry-ruby/pull/1180/files#diff-8f036f2cfa248f9a798edf1f55e2d7f19fbdac08dcc8625c9a980ef8df026d99R52

Reply:
Perhaps we can rename `encode` to `encode_spans` or something similar (if we end up wanting to add `as_emsr` and `encode_metrics` for metrics support)?